File: fsharp/hello-world/HelloWorldTest.fs
module HelloWorldTest

open NUnit.Framework

open HelloWorld

[<Test>]
let ``No name`` () =
    Assert.That(hello None, Is.EqualTo("Hello, World!"))

[<Test>]
let ``Sample name`` () =
    Assert.That(hello (Some "Alice"), Is.EqualTo("Hello, Alice!"))

[<Test>]
let ``Other sample name`` () =
    Assert.That(hello (Some "Bob"), Is.EqualTo("Hello, Bob!"))

File: v1/csharp/hello-world/HelloWorldTest.cs
using NUnit.Framework;

[TestFixture]
public class HelloWorldTest
{
    [Test]
    public void No_name()
    {
        Assert.That(HelloWorld.Hello(null), Is.EqualTo("Hello, World!"));
    }

    [Test]
    public void Sample_name()
    {
        Assert.That(HelloWorld.Hello("Alice"), Is.EqualTo("Hello, Alice!"));
    }

    [Test]
    public void Other_sample_name()
    {
        Assert.That(HelloWorld.Hello("Bob"), Is.EqualTo("Hello, Bob!"));
    }
}
